rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -233,15 +232,25 @@ def _query_deposits_and_rewards(
             )
             proxy_owner = self.ethereum.proxies_inquirer.proxy_to_address.get(current_address)
             if proxy_owner is not None:
-                data[proxy_owner]['proxies'][current_address][key] = AssetBalance(
+                if data[proxy_owner]['proxies'] is None:
+                    # mypy fails to detect that the defaultdict is returning the
+                    # correct type for the not None case
+                    data[proxy_owner]['proxies'] = defaultdict(lambda: defaultdict(AssetBalance))  # type: ignore[arg-type]  # noqa: E501
```

Comment:
The problem here and below is not the None case, but that `defaultdict` needs a callable that doesn't accept any arguments, and here `AssetBalance` is given, which requires `asset` and `balance` arguments to be initialized.

I think that right now it is broken since if at some place there will be access to `data[proxy_owner]['proxies'][some_address].some_property_of_AssetBalance` and `data[proxy_owner]['proxies'][some_address]` won't be defined, it will result in an error.